### PR TITLE
Issue #14938: Updated the google style coverage page regarding JSNI method

### PR DESCRIPTION
**Pull request by Zopsss**
Issue #14938 

Updated google style coverage page, removed the `JSNI could not be detected` message in section `4.4 Column Limit: 100` and added a new message indicating that it will be not supported by us with the issue link which contains all the discussion.

### CODE REVIEW

**Comment by Zopsss** [comment]
GitHub, generate site

**Comment by Zopsss** [comment]
@romani https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/8081036_2024054118/google_style.html#a4.4

File: src/xdocs/google_style.xml [hunk 1]
```diff
@@ -799,17 +799,21 @@
                 <td>
                   <span class="wrapper inline">
                     <img
-                        src="images/ok_blue.png"
```
Comment (rdiachenko) [inline src/xdocs/google_style.xml:802]: I don't think we need to change it to green because we don't cover all Google requirements.

File: src/xdocs/google_style.xml [hunk 2]
```diff
@@ -799,17 +799,21 @@
...
+                        src="images/ok_green.png"
                         alt="" />
                   </span>
                   <a href="checks/sizes/linelength.html#LineLength">LineLength</a>
                   <br />
-                  We can detect URL with protocol type as http://, https:// etc.
```
Comment (rdiachenko) [inline src/xdocs/google_style.xml:807]: This is not related to JSNI. Why did you remove it? Could you recheck please if we cover URLs in tests and update them if it is missed.

Comment (Zopsss) [reply src/xdocs/google_style.xml:807]: I accidentally removed it. But I checked about it, we have support to ignore URLs:

https://github.com/checkstyle/checkstyle/blob/173925e9a52370dd80e66b2e847e1b0ff1b1a7eb/src/main/resources/google_checks.xml#L45-L49

But there is no tests for it. I'll update add tests soon. Thanks for pointing this out :)

File: src/xdocs/google_style.xml (hunk 1 above)
Comment (Zopsss) [reply src/xdocs/google_style.xml:802]: okayy I will revert this 👍

File: src/xdocs/google_style.xml [hunk 3]
```diff
@@ -799,17 +799,21 @@
...
                     JSNI</a>
-                  could not be detected right now, but might be possible after
-                  comments and javadoc support appear in Checkstyle.
+                  is not supported by Checkstyle for this rule.
+                  After careful consideration and discussion,
+                  the Checkstyle maintainers have decided not to implement this
+                  feature in the foreseeable future. For more details regarding this decision,
+                  please refer to the discussion in the following issue:
+                  <a href="https://github.com/checkstyle/checkstyle/issues/14938">
+                    #14938</a>.
```
Comment (romani) [inline src/xdocs/google_style.xml:816]: Too much words.
Please do `JSNI is not supported, see reason at #14938`

**Review by romani — CHANGES_REQUESTED** [review]
Item

File: src/xdocs/google_style.xml (hunk 1 above)
Comment (Zopsss) [reply src/xdocs/google_style.xml]: done

File: src/xdocs/google_style.xml (hunk 3 above)
Comment (Zopsss) [reply src/xdocs/google_style.xml:813]: done

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java [hunk 4]
```diff
@@ -181,6 +181,30 @@ void toManyArgs(int aArg1, int aArg2, int aArg3, int aArg4, int aArg5,
...
+     */
+    public void testHttp() {}
+
+    /**
+     * Very long url with ftp: ftp://ftp.dlink.ru/pub/D-Link_Solutions/D-Link_Solutions_for_Business.pdf
+     */
+    final double testFtp = 99.99;
+
+
+    // Very long url with href: href://ftp.dlink.ru/pub/D-Link_Solutions/D-Link_Solutions_for_Business.pdf
```
Comment (rdiachenko) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java:201]: Please remove this case, it is not a real case.

File: src/xdocs/google_style.xml (hunk 2 above)
Comment (rdiachenko) [reply src/xdocs/google_style.xml:808]: > I accidentally removed it

I still see the line is removed, please put it back.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java (hunk 4 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java:201]: We can keep it and it should be violation as it is unknown protocol "href".
Our config supports only well known

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java:201]: Why is it not a violation?

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java:201]: We support href URLs, as you can see here:

https://github.com/checkstyle/checkstyle/blob/173925e9a52370dd80e66b2e847e1b0ff1b1a7eb/src/main/resources/google_checks.xml#L45-L49

it is used for `<a href="someurl"></a>`

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java:201]: `Very long url with hre` --> `placing 'href' word  in any long line suppress violation `

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java:201]: I see, then we will need to remove `href` from the ignorePattern and also update the `a href` to `<a href="`.

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java]: i have updated the regex, now it will be only valid when it is a html hyper link:

https://github.com/checkstyle/checkstyle/blob/a41e7bffd380c18838ea34ae1513684f8821f8db/src/main/resources/google_checks.xml#L45-L49

**Comment by Zopsss** [comment]
> I still see the line is removed, please put it back.

done.

**Comment by Zopsss** [comment]
> I still see the line is removed, please put it back.

done @rdiachenko

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java (hunk 4 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java]: `|href` was for cases when line is starting by attribute `href="...."`, tag a is on previous line

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java]: Nobody complained about this suppression for 10 years, let's not change it.

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java]: done:
https://github.com/checkstyle/checkstyle/blob/8d93b36bc13a0161531563036cca811fc0a4840c/src/main/resources/google_checks.xml#L45-L50

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java]: after re-adding the `|href`, this line is not considered as violation:

https://github.com/checkstyle/checkstyle/blob/8d93b36bc13a0161531563036cca811fc0a4840c/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java#L201

as you said here: https://github.com/checkstyle/checkstyle/pull/14943#discussion_r1630637409. This line should cause violation but because `|href` is present, it is bypassing the rule.

Failing Test case result:

```
Error message at position 2 of 'expected' does not match actual message
expected: C:\OpenSource\checkstyle\src\it\resources\com\google\checkstyle\test\chapter4formatting\rule44columnlimit\InputLineLength.java:201: Line is longer than 100 characters (found 120).
but was : Audit done.
```

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java]: So should we keep `|href` or not?

File: src/main/resources/google_checks.xml [hunk 5]
```diff
@@ -45,7 +45,8 @@
   <module name="LineLength">
     <property name="fileExtensions" value="java"/>
     <property name="max" value="100"/>
-    <property name="ignorePattern" value="^package.*|^import.*|a href|href|http://|https://|ftp://"/>
+    <property name="ignorePattern"
+             value="^package.*|^import.*|&lt;a href=&quot;(.*?)&quot;&gt;(.*?)&lt;\/a&gt;|href|http://|https://|ftp://"/>
```
Comment (romani) [inline src/main/resources/google_checks.xml:49]: remove `&lt;a ` and `&lt;\/a&gt;`

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java (hunk 4 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java]: ok, remove `|href|`.
BUT lets do this in separatate issue, to let users see that we changed it and it was in releasenotes, not a secret side effect under JSNI.

in this PR lets focus on JSNI only and no other updates to config or mapping table.

File: src/main/resources/google_checks.xml (hunk 5 above)
Comment (romani) [reply src/main/resources/google_checks.xml:49]: https://github.com/checkstyle/checkstyle/pull/14943#discussion_r1631183000

Comment (Zopsss) [reply src/main/resources/google_checks.xml]: reverted all these changes.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java (hunk 4 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule44columnlimit/InputLineLength.java]: separate issue: https://github.com/checkstyle/checkstyle/issues/14955

**Review by rdiachenko — APPROVED** [review]
lgtm

**Review by romani — APPROVED** [review]
